Memory Leak Fix & onPictureSavedListener update

yayaa opened the pull request:
I also encountered with this memory leak [reported here](https://code.google.com/p/android/issues/detail?id=173788) and removed MediaScannerConnection from code and returns image file path instead of uri.

If file requires to be displayed in gallery, below method can be used as well.

```
    public static void scanGallery(Activity activity, File modifiedFile) {
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
                Intent mediaScanIntent = new Intent(
                Intent.ACTION_MEDIA_SCANNER_SCAN_FILE);
                Uri contentUri = Uri.fromFile(modifiedFile);
                mediaScanIntent.setData(contentUri);
                activity.sendBroadcast(mediaScanIntent);
            } else {
                activity.sendBroadcast(new Intent(
                        Intent.ACTION_MEDIA_MOUNTED,
                        Uri.parse("file://" + Environment.getExternalStorageDirectory())));
            }
    }
```

duchuyctlk commented on library/src/jp/co/cyberagent/android/gpuimage/GPUImageView.java:
```diff
@@ -438,32 +439,34 @@ private void saveImage(final String folderName, final String fileName, final Bit
+        void onPictureSaveFailed();
     }
-}
+}
```
You should put an empty line at the end of file.

yayaa commented:
Hey @wasabeef ,

I'm not sure who is in charge of this library anymore, but i saw the last commits are yours, so wanted to try :) I fixed a memoryLeak and black background issue. I believe these stuff are kind of important for the library, so would be nice if you could spare some time and take a look at this PR and merge it. 

- Don't know why did you decide to remove `.travis.yml` file, but without that travis tries to treat the repo as Ruby not Android, so i had to bring that configuration file back.

- Version code automatisation wasn't working correctly, had to revert that back manuel defined in gradle.properties but tried to respect the result you wanted, and updated with latest commit number :)

Please consider `Squash and merge` since i had to try multiple stuff to get travis happy :) and releasing a new version would be really nice!

yayaa commented:
Nobody?

![lonely](https://cloud.githubusercontent.com/assets/2683375/25406998/f9833988-2a08-11e7-94bc-4bebaeecb728.gif)

yayaa commented:
![what_now](https://cloud.githubusercontent.com/assets/2683375/26029009/73044308-382b-11e7-8868-9178034130f9.gif)

yayaa commented:
Ok, if anybody needs this changes can use mine, i'm done here :)
